kafka-native-test-container/src/test/java/com/ozangunalp/kafka/test/container/KafkaNativeContainerIT.java

```diff
@@ -271,4 +310,14 @@ void testKraftClusterOneController() throws Exception {
         }
     }
 
+    MountableFile classpathResourceWithStringsReplaced(String resourceName, Path tempDir, Map<String, String> replacements) throws IOException {
+        MountableFile serverPropertiesFile = MountableFile.forClasspathResource(resourceName);
+        String props = Files.readString(Path.of(serverPropertiesFile.getFilesystemPath()), StandardCharsets.UTF_8);
+        for (Map.Entry<String, String> entry : replacements.entrySet()) {
+            props = props.replaceAll(quote(entry.getKey()), entry.getValue());
+        }
+        Path out = tempDir.resolve("controller_only.properties");
```

Comment:
"controller_only.properties" -> resourceName?

Reply:
the method is removed as part of refactoring withServerProperties to take Transferable. The test can operate on String now without messing with temporary files.